Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -5837,6 +5842,9 @@ void RenderLayer::styleChanged(StyleDifference diff, const RenderStyle* oldStyle
             if (visibilityChanged || oldStyle->isOverflowVisible() != renderer().style().isOverflowVisible())
                 m_scrollableArea->computeHasCompositedScrollableOverflow(diff <= StyleDifference::RepaintLayer ? LayoutUpToDate::Yes : LayoutUpToDate::No);
         }
+
+        if (oldStyle->hasZeroOpacity() != renderer().style().hasZeroOpacity())
+            setNeedsPositionUpdate();
```

Comment:
The naming is odd; opacity has nothing to do with position.

Reply:
It's true, though I think the mismatch isn't here.

`setNeedsPositionUpdate` means 'we need to run `recursiveUpdateLayerPositions` for this layer.

Both are probably misnamed, since 'updating layer positions' also does a bunch of other non-position things, including computing repaint rects (which do depend on opacity).